packages/serialize/src/index.js

```diff
@@ -255,6 +254,74 @@ function handleInterpolation(
     : interpolation
 }
 
+function createStringFromObjectEntry(mergedProps, registered, key, value) {
+  switch (typeof value) {
+    case 'string':
+    case 'number':
+      if (registered != null && registered[value] !== undefined) {
+        return `${key}{${registered[value]}}`
+      } else {
+        return `${processStyleName(key)}:${processStyleValue(key, value)};`
+      }
+    case 'function': {
+      const interpolated = handleInterpolation(
+        mergedProps,
+        registered,
+        value,
+        false
+      )
+
+      return createStringFromObjectEntry(
+        mergedProps,
+        registered,
+        key,
+        interpolated
+      )
+    }
+    case 'object':
+      if (
+        key === 'NO_COMPONENT_SELECTOR' &&
+        process.env.NODE_ENV !== 'production'
+      ) {
+        throw new Error(
+          'Component selectors can only be used in conjunction with babel-plugin-emotion.'
+        )
+      }
+
+      if (Array.isArray(value) && typeof value[0] === 'string') {
```

Comment:
I've removed `&& (registered == null || registered[value[0]] === undefined)` from here. If it was needed - please share an example so I can add a test for this.

Reply:
It's for cases like this

```jsx
import {css} from 'emotion'

css({':hover': [css({color:'green'})]})
```